### PR TITLE
Find secret

### PR DESCRIPTION
**Pull request by matthewjablack**
### Description

This PR adds  `getSecret` for both Ethereum and Bitcoin swap providers based on claim transaction hash

### Submission Checklist :pencil:

- [x] `getSecret` for both Ethereum and Bitcoin swap providers

### CODE REVIEW

File: src/providers/bitcoin/BitcoinSwapProvider.js [hunk 1]
```diff
@@ -161,6 +161,15 @@ export default class BitcoinSwapProvider extends Provider {
     return swapTx ? swapTx.hash : null
   }
 
+  async getSecret (claimTxHash) {
+    const claimTxRaw = await this.getMethod('getRawTransactionByHash')(claimTxHash)
+    const claimTx = await this.getMethod('splitTransaction')(claimTxRaw, true)
```
Comment (monokh) [inline src/providers/bitcoin/BitcoinSwapProvider.js]: you could also just use rpc `decoderawtransaction`

Comment (matthewjablack) [reply src/providers/bitcoin/BitcoinSwapProvider.js]: Good point, there shouldn't be dependency on `BitcoinLedgerProvider` for something like this.

Comment (matthewjablack) [reply src/providers/bitcoin/BitcoinSwapProvider.js]: Updated to use `BitcoinRPCProvider`

File: src/providers/ethereum/EthereumMetaMaskProvider.js [hunk 2]
```diff
@@ -56,12 +56,13 @@ export default class EthereumMetaMaskProvider extends Provider {
     return addresses[0]
   }
 
-  async signMessage (message, from) {
-    from = String(from)
-
+  async signMessage (message) {
```
Comment (harshjv) [inline src/providers/ethereum/EthereumMetaMaskProvider.js:59]: This makes sense for MM. Nice.

Comment (harshjv) [reply src/providers/ethereum/EthereumMetaMaskProvider.js:59]: However, if addresses are being managed by the UI, can we assume that UI will send correct address with `signMessage`?

**Review by harshjv — APPROVED** [review]
Small comments. Otherwise, LGTM

**Comment by monokh** [comment]
In the interest of not having too many branching going on and since you are away @mattBlackDesign, I'm going to merge this so I can follow up with my work cleanly. Feel free to come back to this after you are back :)

**Pull request merged**